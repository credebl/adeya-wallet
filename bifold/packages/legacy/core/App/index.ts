/* eslint-disable import/no-cycle */
import type { OnboardingStyleSheet } from './screens/Onboarding'

import { Agent } from '@aries-framework/core'
import AgentProvider from '@aries-framework/react-hooks'

import indyLedgers from '../configs/ledgers/indy'

import * as components from './components'
import Button, { ButtonType } from './components/buttons/Button'
import CheckBoxRow from './components/inputs/CheckBoxRow'
import CredentialCard from './components/misc/CredentialCard'
import InfoBox, { InfoBoxType } from './components/misc/InfoBox'
import ErrorModal from './components/modals/ErrorModal'
import NetInfo from './components/network/NetInfo'
import Record from './components/record/Record'
import InfoTextBox from './components/texts/InfoTextBox'
import { ToastType } from './components/toast/BaseToast'
import toastConfig from './components/toast/ToastConfig'
import { AttachTourStep } from './components/tour/AttachTourStep'
import { homeTourSteps } from './components/tour/HomeTourSteps'
import { TourBox } from './components/tour/TourBox'
import HomeContentView from './components/views/HomeContentView'
import * as contexts from './contexts'
import { AuthProvider } from './contexts/auth'
import { CommonUtilProvider } from './contexts/commons'
import { NetworkProvider } from './contexts/network'
import { useTour } from './contexts/tour/tour-context'
import { TourProvider } from './contexts/tour/tour-provider'
import { defaultConfiguration } from './defaultConfiguration'
import RootStack from './navigators/RootStack'
import AttemptLockout from './screens/AttemptLockout'
import Developer from './screens/Developer'
import OnboardingPages from './screens/OnboardingPages'
import Splash from './screens/Splash'
import Terms from './screens/Terms'
import UseBiometry from './screens/UseBiometry'
import * as types from './types'

export { LocalStorageKeys } from './constants'
export { initLanguages, initStoredLanguage, translationResources } from './localization'
export { ConfigurationProvider, useConfiguration } from './contexts/configuration'
export { defaultState, mergeReducers, StoreProvider, StoreContext, useStore } from './contexts/store'
export { default as Store, DispatchAction, reducer } from './contexts/reducers/store'
export { Assets as ImageAssets } from './theme'
export { ThemeProvider, useTheme } from './contexts/theme'
export { AnimatedComponentsProvider, useAnimatedComponents } from './contexts/animated-components'
export { ColorPallet } from './theme'
export { animatedComponents } from './animated-components'
export { theme } from './theme'
export { useAuth } from './contexts/auth'
export { NavigationTheme } from './theme'
export { testIdWithKey } from './utils/testable'
export { Screens, Stacks, TabStacks } from './types/navigators'
export { createStyles } from './screens/OnboardingPages'
export { statusBarStyleForColor, StatusBarStyles } from './utils/luminance'
export { BifoldError } from './types/error'
export { EventTypes } from './constants'

export type { AnimatedComponents } from './animated-components'
export type {
  ISVGAssets,
  IFontAttributes,
  IInputAttributes,
  IInputs,
  ITextTheme,
  IBrandColors,
  ISemanticColors,
  INotificationColors,
  IGrayscaleColors,
  IColorPallet,
  IAssets,
  ITheme,
} from './theme'
export type { ConfigurationContext } from './contexts/configuration'
export type { TourStep } from './contexts/tour/tour-context'
export type { GenericFn } from './types/fn'
export type { AuthenticateStackParams } from './types/navigators'
export type { OnboardingStyleSheet }
export type { WalletSecret } from './types/security'
export type { ReducerAction } from './contexts/reducers/store'
export type {
  State,
  Onboarding as OnboardingState,
  LoginAttempt as LoginAttemptState,
  Preferences as PreferencesState,
  Tours as ToursState,
} from './types/state'
export * from '../verifier'

export {
  indyLedgers,
  Agent,
  CommonUtilProvider,
  AgentProvider,
  AuthProvider,
  NetworkProvider,
  TourProvider,
  useTour,
  AttachTourStep,
  TourBox,
  homeTourSteps,
  Button,
  ButtonType,
  CheckBoxRow,
  CredentialCard,
  ErrorModal,
  InfoTextBox,
  InfoBox,
  InfoBoxType,
  ToastType,
  toastConfig,
  RootStack,
  NetInfo,
  OnboardingPages,
  Splash,
  Developer,
  Terms,
  HomeContentView,
  UseBiometry,
  AttemptLockout,
  Record,
  defaultConfiguration,
  types,
  components,
  contexts,
}
