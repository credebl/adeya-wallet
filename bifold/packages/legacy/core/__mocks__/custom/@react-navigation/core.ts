const navigate = jest.fn()

const navigation = {
  navigate,
  setOptions: jest.fn(),
  getParent: jest.fn(() => ({
    navigate,
  })),
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

export { useNavigation, useIsFocused }
