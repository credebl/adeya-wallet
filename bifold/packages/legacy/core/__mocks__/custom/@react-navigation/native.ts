const useIsFocused = jest.fn()
const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()

export { useIsFocused, useFocusEffect, createNavigatorFactory }
