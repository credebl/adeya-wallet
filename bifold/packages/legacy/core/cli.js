/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable no-undef */
/* eslint-disable no-console */

function syncPackageJson() {
  const { execSync } = require('child_process')
  const pak = require('./package.json')
  function parseNameAndVersion(val) {
    const sep = val.lastIndexOf('@')
    return [val.substring(0, sep), val.substring(sep + 1)]
  }
  function npmInstall(command) {
    console.log(`cmd\n${command}`)
    execSync(command, { stdio: 'inherit', cwd: process.cwd() })
  }
  const lockedDependencies = execSync('npm ls --package-lock-only', { cwd: __dirname })
    .toString()
    .trim()
    .split('\n')
    .splice(1)
    .map((item) => {
      return item.substring(4)
    })
    .reduce((prev, val) => {
      const [name, version] = parseNameAndVersion(val)
      prev[name] = version
      return prev
    }, {})

  //const peerDependencies = pak.peerDependencies
  //console.log(`peerDependencies:${ Object.keys(pak.peerDependencies).length}`)
  for (const item of Object.keys(pak.peerDependencies)) {
    pak.peerDependencies[item] = '' + lockedDependencies[item]
  }
  console.log(JSON.stringify(pak.peerDependencies, undefined, 2))

  //console.log(`devDependencies:${ Object.keys(pak.devDependencies).length}`)
  for (const item of Object.keys(pak.devDependencies)) {
    if (pak.devDependencies[item].startsWith('npm:')) {
      const curVersion = pak.devDependencies[item]
      const [name] = parseNameAndVersion(curVersion)
      pak.devDependencies[item] = name + '@' + lockedDependencies[item + '@' + name]
    } else {
      pak.devDependencies[item] = '' + lockedDependencies[item]
    }
  }
  console.log('Installing Dependencies ...')
  for (const item of Object.keys(pak.peerDependencies)) {
    npmInstall(`npm install ${item}@${pak.peerDependencies[item]} --force --save-exact`, {
      stdio: 'inherit',
      cwd: process.cwd(),
    })
  }
  console.log('Installing Dev Dependencies ...')
  const devAllowList = [
    '@babel/core',
    '@babel/runtime',
    '@types/lodash.merge',
    '@types/react',
    '@types/react-native',
    'babel-plugin-module-resolver',
    'metro-react-native-babel-preset',
    'react-native-svg-transformer',
    'typescript',
  ]
  for (const item of Object.keys(pak.devDependencies)) {
    if (devAllowList.includes(item)) {
      npmInstall(`npm install -D ${item}@${pak.devDependencies[item]} --force --save-exact`, {
        stdio: 'inherit',
        cwd: process.cwd(),
      })
    }
  }
  //console.log(JSON.stringify(pak.devDependencies, undefined, 2))
  /*
  fs.writeFile(
    'app-package.json',
    JSON.stringify({ dependencies: pak.peerDependencies, devDependencies: pak.devDependencies }, undefined, 2),
    'utf8',
    (err) => {
      if (err) {
        console.log('An error occured while writing JSON Object to File.')
        return console.log(err)
      }
      console.log('JSON file has been saved.')
    }
  )
  */
}
module.exports = {
  cli: (_args) => {
    const args = _args.splice(2)
    console.log(args)
    if (args[0] === 'sync-package-json') {
      syncPackageJson()
    }
  },
}
