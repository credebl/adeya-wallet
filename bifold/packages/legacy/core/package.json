{
  "name": "aries-bifold",
  "version": "0.0.1",
  "scripts": {
    "style-check": "npm run prettier -- --check",
    "style-format": "npm run prettier -- --write",
    "style-watch": "onchange '**/*.+(js|jsx|json|ts|tsx|md|yml|yaml)' -- prettier --write {{changed}}",
    "android": "react-native run-android",
    "ios": "npx react-native run-ios --device",
    "start": "npx react-native start --reset-cache",
    "test": "TZ=GMT jest",
    "test:watch": "TZ=GMT jest --watch --notify --notifyMode=change --coverage",
    "test:coverage": "TZ=GMT jest --coverage",
    "lint": "eslint --ext .js,.jsx,.ts,.tsx --color .",
    "prettier": "prettier --ignore-path .gitignore '**/*.+(js|jsx|json|ts|tsx|md|yml|yaml)'",
    "postinstall": "npx patch-package",
    "pre-commit-lint": "npx lint-staged",
    "setup": "cd .. && husky install .husky",
    "prepack": "bob build",
    "typecheck": "tsc --noEmit",
    "prepare": "install-self-peers -- --ignore-scripts"
  },
  "peerDependencies": {
    "@aries-framework/core": "^0.3.3",
    "@aries-framework/react-hooks": "^0.4.2",
    "@aries-framework/react-native": "^0.3.2",
    "@formatjs/intl-datetimeformat": "^4.2.3",
    "@formatjs/intl-displaynames": "^5.2.3",
    "@formatjs/intl-getcanonicallocales": "^1.7.3",
    "@formatjs/intl-listformat": "^6.3.3",
    "@formatjs/intl-locale": "^2.4.37",
    "@formatjs/intl-numberformat": "^7.2.3",
    "@formatjs/intl-pluralrules": "^4.1.3",
    "@formatjs/intl-relativetimeformat": "^9.2.3",
    "@react-native-async-storage/async-storage": "^1.12.1",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/netinfo": "^9.3.0",
    "@react-navigation/bottom-tabs": "^6.0.9",
    "@react-navigation/core": "^6.1.0",
    "@react-navigation/devtools": "^6.0.8",
    "@react-navigation/native": "^6.0.6",
    "@react-navigation/stack": "^6.0.11",
    "axios": "^0.21.0",
    "i18next": "^21.4.0",
    "indy-sdk-react-native": "^0.2.2",
    "lodash.flatten": "^4.4.0",
    "lodash.startcase": "^4.4.0",
    "moment": "^2.29.4",
    "query-string": "^7.0.1",
    "react": "17.0.2",
    "react-i18next": "^11.13.0",
    "react-native": "0.66.5",
    "react-native-animated-pagination-dots": "^0.1.72",
    "react-native-argon2": "^2.0.1",
    "react-native-bouncy-checkbox": "^3.0.5",
    "react-native-camera": "^3.31.1",
    "react-native-collapsible": "^1.6.0",
    "react-native-config": "^1.4.2",
    "react-native-confirmation-code-field": "^7.3.0",
    "react-native-device-info": "^8.4.9",
    "react-native-encrypted-storage": "^4.0.2",
    "react-native-fs": "^2.16.6",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-get-random-values": "^1.7.0",
    "react-native-gifted-chat": "^0.16.3",
    "react-native-keychain": "^8.1.1",
    "react-native-localize": "^2.2.4",
    "react-native-permissions": "^3.6.1",
    "react-native-qrcode-svg": "^6.0.6",
    "react-native-reanimated": "^2.2.4",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-screens": "^3.3.0",
    "react-native-splash-screen": "^3.2.0",
    "react-native-svg": "^12.1.1",
    "react-native-tcp-socket": "^6.0.3",
    "react-native-toast-message": "^2.0.1",
    "react-native-uuid": "^2.0.1",
    "react-native-vector-icons": "^8.1.0",
    "uuid": "^8.1.0"
  },
  "devDependencies": {
    "@aries-framework/core": "^0.3.3",
    "@aries-framework/react-hooks": "^0.3.0",
    "@aries-framework/react-native": "^0.2.5",
    "@babel/core": "^7.12.9",
    "@babel/runtime": "^7.12.5",
    "@commitlint/cli": "^11.0.0",
    "@formatjs/intl-datetimeformat": "^4.2.3",
    "@formatjs/intl-displaynames": "^5.2.3",
    "@formatjs/intl-getcanonicallocales": "^1.7.3",
    "@formatjs/intl-listformat": "^6.3.3",
    "@formatjs/intl-locale": "^2.4.37",
    "@formatjs/intl-numberformat": "^7.2.3",
    "@formatjs/intl-pluralrules": "^4.1.3",
    "@formatjs/intl-relativetimeformat": "^9.2.3",
    "@react-native-async-storage/async-storage": "^1.12.1",
    "@react-native-community/eslint-config": "^2.0.0",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/netinfo": "^9.3.0",
    "@react-navigation/bottom-tabs": "^6.0.9",
    "@react-navigation/core": "^6.1.0",
    "@react-navigation/devtools": "^6.0.8",
    "@react-navigation/native": "^6.0.6",
    "@react-navigation/stack": "^6.0.11",
    "@team-griffin/install-self-peers": "^1.1.1",
    "@testing-library/jest-native": "^4.0.13",
    "@testing-library/react-native": "^9.0.0",
    "@types/indy-sdk-react-native": "npm:@types/indy-sdk@^1.16.18",
    "@types/jest": "^27.4.0",
    "@types/lodash.flatten": "^4.4.6",
    "@types/lodash.startcase": "^4.4.6",
    "@types/react": "^16.14.4",
    "@types/react-native": "^0.63.50",
    "@types/react-native-vector-icons": "^6.4.6",
    "@types/react-test-renderer": "^16.9.5",
    "@typescript-eslint/eslint-plugin": "^4.20.0",
    "@typescript-eslint/parser": "^4.33.0",
    "axios": "^0.21.0",
    "babel-jest": "^27.5.1",
    "babel-plugin-module-resolver": "^4.1.0",
    "commitlint": "^11.0.0",
    "eslint": "^7.14.0",
    "eslint-import-resolver-typescript": "^2.5.0",
    "eslint-plugin-import": "^2.25.2",
    "eslint-plugin-prettier": "^3.3.1",
    "husky": "^7.0.0",
    "i18next": "^21.4.0",
    "indy-sdk-react-native": "^0.2.2",
    "install-peerdeps": "^3.0.3",
    "jest": "^27.5.1",
    "lint-staged": "^11.2.6",
    "lodash.flatten": "^4.4.0",
    "lodash.startcase": "^4.4.0",
    "metro-react-native-babel-preset": "^0.66.2",
    "mockdate": "^3.0.5",
    "moment": "^2.29.4",
    "node-notifier": "^10.0.1",
    "patch-package": "^6.4.4",
    "prettier": "^2.1.2",
    "query-string": "^7.0.1",
    "react": "17.0.2",
    "react-i18next": "^11.13.0",
    "react-native": "0.66.5",
    "react-native-animated-pagination-dots": "^0.1.72",
    "react-native-argon2": "^2.0.1",
    "react-native-bouncy-checkbox": "^3.0.5",
    "react-native-builder-bob": "^0.18.2",
    "react-native-camera": "^3.31.1",
    "react-native-collapsible": "^1.6.0",
    "react-native-config": "^1.4.2",
    "react-native-confirmation-code-field": "^7.3.0",
    "react-native-device-info": "^8.4.9",
    "react-native-encrypted-storage": "^4.0.2",
    "react-native-fs": "^2.16.6",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-get-random-values": "^1.7.0",
    "react-native-gifted-chat": "^0.16.3",
    "react-native-keychain": "^8.1.1",
    "react-native-localize": "^2.1.5",
    "react-native-permissions": "^3.6.1",
    "react-native-qrcode-svg": "^6.0.6",
    "react-native-reanimated": "^2.2.4",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-scalable-image": "^1.1.0",
    "react-native-screens": "^3.3.0",
    "react-native-splash-screen": "^3.2.0",
    "react-native-svg": "^12.1.1",
    "react-native-svg-transformer": "^0.14.3",
    "react-native-tcp-socket": "^6.0.3",
    "react-native-toast-message": "^2.0.1",
    "react-native-uuid": "^2.0.1",
    "react-native-vector-icons": "^8.1.0",
    "react-test-renderer": "17.0.2",
    "typescript": "4.4.3",
    "uuid": "^8.1.0"
  },
  "main": "lib/commonjs/index.js",
  "bin": {
    "bifold": "bin/bifold"
  },
  "react-native": "App/index.ts",
  "module": "lib/module/index.js",
  "types": "lib/typescript/App/index.d.ts",
  "files": [
    "App",
    "lib",
    "!**/__tests__",
    "!**/__fixtures__",
    "!**/__mocks__"
  ],
  "react-native-builder-bob": {
    "source": "App",
    "output": "lib",
    "targets": [
      "commonjs",
      "module",
      "typescript"
    ]
  },
  "installConfig": {
    "hoistingLimits": "workspaces"
  }
}
