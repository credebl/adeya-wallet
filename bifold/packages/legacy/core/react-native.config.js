module.exports = {
  assets: ['./App/assets/fonts'],
}
