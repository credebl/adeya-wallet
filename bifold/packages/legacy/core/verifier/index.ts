export { ProofRequestType } from './types/proof-reqeust-template'
export type {
  ProofRequestTemplate,
  DifProofRequestTemplatePayload,
  DifProofRequestTemplatePayloadData,
  IndyProofRequestTemplatePayload,
  IndyProofRequestTemplatePayloadData,
  IndyRequestedAttribute,
  IndyRequestedPredicate,
} from './types/proof-reqeust-template'
export type {
  GroupedSharedProofData,
  GroupedSharedProofDataItem,
  CredentialSharedProofData,
  ParsedIndyProof,
  UnresolvedPredicate,
  ResolvedPredicate,
  SharedAttributesGroup,
  SharedGroupedAttribute,
  SharedAttribute,
  MissingAttribute,
} from './types/proof'
export type { ProofCustomMetadata } from './types/metadata'
export { ProofMetadata } from './types/metadata'
export { useProofsByTemplateId } from './hooks/proofs'
export {
  getProofIdentifiers,
  parseIndyProof,
  groupSharedProofDataByCredential,
  getProofData,
  isPresentationReceived,
  isPresentationFailed,
  markProofAsViewed,
  linkProofWithTemplate,
} from './utils/proof'
export type { CreateProofRequestInvitationResult, SendProofRequestResult } from './utils/proof-request'
export {
  findProofRequestMessage,
  buildProofRequestDataForTemplate,
  createConnectionlessProofRequestInvitation,
  sendProofRequest,
  hasPredicates,
  isParameterizable,
} from './utils/proof-request'
export { defaultProofRequestTemplates } from './request-templates'
