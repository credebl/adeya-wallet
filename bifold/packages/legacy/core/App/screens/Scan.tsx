import type { BarCodeReadEvent } from 'react-native-camera'

import { useAgent } from '@aries-framework/react-hooks'
import { StackScreenProps } from '@react-navigation/stack'
import React, { useState, useEffect } from 'react'
import { useTranslation } from 'react-i18next'
import { Platform } from 'react-native'
import { check, Permission, PERMISSIONS, request, RESULTS } from 'react-native-permissions'
import Toast from 'react-native-toast-message'

import QRScanner from '../components/misc/QRScanner'
import CameraDisclosureModal from '../components/modals/CameraDisclosureModal'
import LoadingModal from '../components/modals/LoadingModal'
import { ToastType } from '../components/toast/BaseToast'
import { BifoldError, QrCodeScanError } from '../types/error'
import { ConnectStackParams, Screens, Stacks } from '../types/navigators'
import { PermissionContract } from '../types/permissions'
import { connectFromInvitation, getJson, getUrl, receiveMessageFromUrlRedirect } from '../utils/helpers'

export type ScanProps = StackScreenProps<ConnectStackParams>

const Scan: React.FC<ScanProps> = ({ navigation }) => {
  const { agent } = useAgent()
  const { t } = useTranslation()
  const [loading, setLoading] = useState<boolean>(true)
  const [showDisclosureModal, setShowDisclosureModal] = useState<boolean>(true)
  const [qrCodeScanError, setQrCodeScanError] = useState<QrCodeScanError | null>(null)

  const handleInvitation = async (value: string): Promise<void> => {
    try {
      const connectionRecord = await connectFromInvitation(value, agent)
      navigation.getParent()?.navigate(Stacks.ConnectionStack, {
        screen: Screens.Connection,
        params: { connectionId: connectionRecord.id },
      })
    } catch (err: unknown) {
      try {
        // if scanned value is json -> pass into AFJ as is
        const json = getJson(value)
        if (json) {
          await agent?.receiveMessage(json)
          navigation.getParent()?.navigate(Stacks.ConnectionStack, {
            screen: Screens.Connection,
            params: { threadId: json['@id'] },
          })
          return
        }

        // if scanned value is url -> receive message from it
        const url = getUrl(value)
        if (url) {
          const message = await receiveMessageFromUrlRedirect(value, agent)
          navigation.getParent()?.navigate(Stacks.ConnectionStack, {
            screen: Screens.Connection,
            params: { threadId: message['@id'] },
          })
          return
        }
      } catch (err: unknown) {
        const error = new BifoldError(t('Error.Title1031'), t('Error.Message1031'), (err as Error).message, 1031)
        throw error
      }
    }
  }

  const handleCodeScan = async (event: BarCodeReadEvent) => {
    setQrCodeScanError(null)
    try {
      const uri = event.data
      await handleInvitation(uri)
    } catch (e: unknown) {
      const error = new QrCodeScanError(t('Scan.InvalidQrCode'), event.data)
      setQrCodeScanError(error)
    }
  }

  const permissionFlow = async (method: PermissionContract, permission: Permission): Promise<boolean> => {
    try {
      const permissionResult = await method(permission)
      if (permissionResult === RESULTS.GRANTED) {
        setShowDisclosureModal(false)
        return true
      }
    } catch (error: unknown) {
      Toast.show({
        type: ToastType.Error,
        text1: t('Global.Failure'),
        text2: (error as Error)?.message || t('Error.Unknown'),
        visibilityTime: 2000,
        position: 'bottom',
      })
    }

    return false
  }

  const requestCameraUse = async (): Promise<boolean> => {
    if (Platform.OS === 'android') {
      return await permissionFlow(request, PERMISSIONS.ANDROID.CAMERA)
    } else if (Platform.OS === 'ios') {
      return await permissionFlow(request, PERMISSIONS.IOS.CAMERA)
    }

    return false
  }

  useEffect(() => {
    const asyncEffect = async () => {
      if (Platform.OS === 'android') {
        await permissionFlow(check, PERMISSIONS.ANDROID.CAMERA)
      } else if (Platform.OS === 'ios') {
        await permissionFlow(check, PERMISSIONS.IOS.CAMERA)
      }
      setLoading(false)
    }

    asyncEffect()
  }, [])

  if (loading) {
    return <LoadingModal />
  }

  if (showDisclosureModal) {
    return <CameraDisclosureModal requestCameraUse={requestCameraUse} />
  }

  return <QRScanner handleCodeScan={handleCodeScan} error={qrCodeScanError} enableCameraOnError={true} />
}

export default Scan
