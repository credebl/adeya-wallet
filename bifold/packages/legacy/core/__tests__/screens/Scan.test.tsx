import { useNavigation } from '@react-navigation/core'
import { render, waitFor } from '@testing-library/react-native'
import React from 'react'

import Scan from '../../App/screens/Scan'

jest.mock('react-native-permissions', () => require('react-native-permissions/mock'))
jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

describe('Scan Screen', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('Renders correctly', async () => {
    const tree = render(<Scan navigation={useNavigation()} route={{} as any} />)
    await waitFor(
      () => {
        expect(tree).toMatchSnapshot()
      },
      { timeout: 10000 }
    )
  })
})
