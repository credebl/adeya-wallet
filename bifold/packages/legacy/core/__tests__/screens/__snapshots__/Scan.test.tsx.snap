// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scan Screen Renders correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "flexDirection": "column",
          "justifyContent": "center",
          "minHeight": 1334,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
      testID="com.ariesbifold:id/LoadingActivityIndicator"
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": "33%",
            "width": "33%",
          }
        }
        testID="com.ariesbifold:id/LoadingActivityIndicatorImage"
      />
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "position": "absolute",
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <
          fill="#FFFFFF"
          height={200}
          width={200}
        />
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
